### PR TITLE
DM-34481: Fix cp_pipe fringe PipelineDataCycleError

### PR DESCRIPTION
**Pull request by czwa**
Discard the correct input for fringes.

### CODE REVIEW

File: python/lsst/ip/isr/isrTask.py [hunk 1]
```diff
@@ -287,7 +287,7 @@ def __init__(self, *, config=None):
         if config.doFlat is not True:
             self.prerequisiteInputs.discard("flat")
         if config.doFringe is not True:
-            self.prerequisiteInputs.discard("fringe")
+            self.prerequisiteInputs.discard("fringes")
```
Comment (PaulPrice) [inline python/lsst/ip/isr/isrTask.py:290]: `self.prerequisiteInputs.remove` might be more suitable: it detects typos.

**Review by PaulPrice — APPROVED** [review]

File: python/lsst/ip/isr/isrTask.py (hunk 1 above)
Comment (erykoff) [reply python/lsst/ip/isr/isrTask.py:290]: That's what I said on slack!  All of these here should be `remove` instead of `discard`.

**Review by PaulPrice — APPROVED** [review]
💯

File: python/lsst/ip/isr/isrTask.py (hunk 1 above)
Comment (czwa) [reply python/lsst/ip/isr/isrTask.py]: I've switched them all, so this shouldn't come up again.

**Pull request merged**